### PR TITLE
Update for dot hpp in v1/3

### PR DESCRIPTION
**Pull request by james-d-mitchell**
This is done modulo finishing and merging the corresponding PR in libsemigroups.

### CODE REVIEW

File: docs/source/data-structures/visualisation/index.rst [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+Functions for creating Dot objects
+----------------------------------
+
+The function ``dot`` can be used to produce initial :any:`Dot` objects for
+various of the data structures and algorithms in libsemigroups:
+
+.. currentmodule:: _libsemigroups_pybind11
+
+* :any:`dot(AhoCorasick)`
```
Comment (Joseph-Edwards) [inline docs/source/data-structures/visualisation/index.rst:31]:
```suggestion
* :any:`aho_corasick.dot`
```
The original gives
```
WARNING: 'any' reference target not found: dot(AhoCorasick)
```

File: src/aho-corasick.cpp [hunk 2]
```diff
@@ -427,6 +428,9 @@ Traverse the trie of *ac* from the root using suffix links where necessary.
 
 This function performs the same as ``traverse_word(ac, AhoCorasick.root, w)``
 )pbdoc");
+
+    // TODO(doc)
+    m.def("dot", [](AhoCorasick& ac) { return aho_corasick::dot(ac); });
```
Comment (Joseph-Edwards) [inline src/aho-corasick.cpp:433]: Do we want to call this `aho_corasick_dot` to simulate the namespace, and then `from ... import aho_corasick_dot as dot`? At the moment, the `AhoCorasick` doc contains the Dot doc for both `WordGraph` as well:
image

File: src/word-graph.cpp [hunk 3]
```diff
@@ -528,5 +528,8 @@ Find the node that a path starting at a given node leads to.
               &to_word_graph<node_type>),
           py::arg("num_nodes"),
           py::arg("l"));
+    // TODO(doc)
+    m.def("dot",
+          [](WordGraph<node_type> const& wg) { return word_graph::dot(wg); });
```
Comment (Joseph-Edwards) [inline src/word-graph.cpp:533]: Same as above, but with `word_graph_dot`?

**Review by Joseph-Edwards — COMMENTED** [review]
Modulo my few comments, this looks good to me

File: src/aho-corasick.cpp (hunk 2 above)
Comment (james-d-mitchell) [reply src/aho-corasick.cpp]: Done

File: src/word-graph.cpp [hunk 4]
```diff
@@ -528,5 +528,26 @@ Find the node that a path starting at a given node leads to.
               &to_word_graph<node_type>),
           py::arg("num_nodes"),
           py::arg("l"));
+    // TODO(doc)
```
Comment (james-d-mitchell) [inline src/word-graph.cpp:531]:
```suggestion
```

File: src/aho-corasick.cpp [hunk 5]
```diff
@@ -426,6 +427,22 @@ This function performs the same as ``traverse_word(ac, AhoCorasick.root, w)``
...
 This function performs the same as ``traverse_word(ac, AhoCorasick.root, w)``
+)pbdoc");
+
+    m.def(
+        "aho_corasick_dot",
+        [](AhoCorasick& ac) { return aho_corasick::dot(ac); },
+        py::arg("ac"),
+        R"pbdoc(
+:sig=(ac: AhoCorasick) -> Dot:
+::only-document-once:
```
Comment (Joseph-Edwards) [inline src/aho-corasick.cpp:438]:
```suggestion
```
We don't need this here, since nothing is being overloaded

**Comment by Joseph-Edwards** [comment]
We should also add `dot` to the list of helper functions in `ac-helpers.rst`:
```
  .. autosummary::
     :nosignatures:
  
     add_word
     rm_word
+    dot
     traverse_word
```

File: src/aho-corasick.cpp (hunk 5 above)
Comment (james-d-mitchell) [reply src/aho-corasick.cpp]: Tah, forgot to push :)

**Comment by james-d-mitchell** [comment]
I think the ci will fail here until #180  is merged, since we've merged the forest-hpp pr into libsemigroups already.

**Comment by james-d-mitchell** [comment]
@Joseph-Edwards should be good to go

**Pull request merged**